bfxapi/rest/bfx_rest.py

```diff
@@ -630,6 +754,97 @@ async def submit_update_order(self, orderId, price=None, amount=None, delta=None
 
+    async def get_auth_pulse_hist(self, is_public=None):
+        """
+        Allows you to retrieve your private pulse history or the public pulse history with an additional UID_LIKED field.
+
+        # Attributes
+        @param is_public int: allows you to receive the public pulse history with the UID_LIKED field
+        @return Array [ PID, MTS, _PLACEHOLDER, PUID, _PLACEHOLDER, TITLE,
+        CONTENT, _PLACEHOLDER, _PLACEHOLDER, IS_PIN, IS_PUBLIC, COMMENTS_DISABLED,
+        TAGS, META,LIKES, UID_LIKED, _PLACEHOLDER, [ PUID, MTS, _PLACEHOLDER, NICKNAME,
+        _PLACEHOLDER, PICTURE, TEXT, _PLACEHOLDER, _PLACEHOLDER, TWITTER_HANDLE, _PLACEHOLDER,
+        FOLLOWERS, FOLLOWING, _PLACEHOLDER, _PLACEHOLDER, _PLACEHOLDER, TIPPING_STATUS ], COMMENTS,
+        _PLACEHOLDER, _PLACEHOLDER ]
+        """
+        endpoint = f"auth/r/pulse/hist"
+        if is_public:
+            endpoint += f'?isPublic={is_public}'
+        hist = await self.post(endpoint)
+        return hist
+
+    async def pulse_add(self, title, content, parent=None, is_pin=False,
+                        attachments=[], disable_comments=False, is_public=True):
+        """
+        Allows you to write Pulse messages
+
+        # Attributes
+        @param title str: title of the message (min 16, max 120 characters)
+        @param content str: content of the message
+        @param parent str: Pulse Message ID (PID) of parent post
+        @param is_pin boolean: is message pinned?
+        @param attachments list of str: base64 format
+        @param disable_comments boolean: are comments disabled?
+        @param is_public boolean: is a public message?
+        @return Array [ PID, MTS, _PLACEHOLDER, PUID, _PLACEHOLDER, TITLE,
+        CONTENT, _PLACEHOLDER, _PLACEHOLDER, IS_PIN, IS_PUBLIC, COMMENTS_DISABLED,
+        TAGS // This inner array contains zero or more tag strings ATTACHMENTS, _PLACEHOLDER,
+        LIKES, UID_LIKED, _PLACEHOLDER, [], ... ]
+        """
+        endpoint = f"auth/w/pulse/add"
+        payload = {
+            'title': title,
+            'content': content,
+            'isPin': 1 if is_pin else 0,
+            'attachments': attachments,
+            'disableComments': 1 if disable_comments else 0,
+            'isPublic': 1 if is_public else 0
+        }
+        if parent:
+            payload['parent'] = parent
+        message = await self.post(endpoint, payload)
+        return message
+
+    async def pulse_delete(self, pid):
+        """
+        Allows you to delete your Pulse messages
+
+        # Attributes
+        @param pid str: ID of the pulse message that you want to delete
+        @return Array [1] or [0]
+        """
+        endpoint = f"auth/w/pulse/del"
+        payload = {
+            'pid': pid
+        }
+        message = await self.post(endpoint, payload)
+        return message
+
+    async def generate_invoice(self, wallet='exchange', currency='LNX', amount='0.02'):
```

Comment:
```
amount='0.02'
```

amount is optional, so probably a default does not make sense here

Reply:
amount is not optional, there was an error in the docs (https://docs.bitfinex.com/reference#rest-auth-deposit-invoice).